tasks/ServiceFabricDockerPush/task.json

```diff
@@ -0,0 +1,128 @@
+    "name": "ServiceFabricDockerPush",
+    "friendlyName": "Push Docker Images for Service Fabric",
+    "description": "Push Docker images associated with a Service Fabric application. Task can be used with Docker or Azure Container registry.",
+    "helpMarkDown": "[More Information](https://go.microsoft.com/fwlink/?linkid=848006)",
+    "category": "Build",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 1,
+        "Patch": 0
+    },
+    "demands": [],
+    "preview": "true",
+    "groups": [
+        {
+            "name": "advanced",
+            "displayName": "Advanced Options",
+            "isExpanded": false
+        }
+    ],
+    "inputs": [
+        {
+            "name": "containerregistrytype",
+            "type": "pickList",
+            "label": "Container Registry Type",
+            "defaultValue": "Azure Container Registry",
+            "required": true,
+            "options": {
+                "Azure Container Registry": "Azure Container Registry",
+                "Container Registry": "Container Registry"
+            },
+            "helpMarkDown": "Select a Container Registry Type."
+        },
+        {
+            "name": "dockerRegistryEndpoint",
+            "type": "connectedService:dockerregistry",
+            "label": "Docker Registry Connection",
+            "helpMarkDown": "Select a Docker registry connection. Required for commands that need to authenticate with a registry.",
+            "visibleRule": "containerregistrytype = Container Registry"
+        },
+        {
+            "name": "azureSubscriptionEndpoint",
+            "type": "connectedService:AzureRM",
+            "label": "Azure subscription",
+            "helpMarkDown": "Select an Azure subscription",
+            "visibleRule": "containerregistrytype = Azure Container Registry"
+        },
+        {
+            "name": "azureContainerRegistry",
+            "label": "Azure Container Registry",
+            "type": "pickList",
+            "helpMarkDown": "Select an Azure Container Registry",
+            "visibleRule": "containerregistrytype = Azure Container Registry",
+            "defaultValue": ""
+        },
+        {
+            "name": "applicationPackagePath",
+            "type": "filePath",
+            "label": "Service Fabric Application Package Path",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Path to the Service Fabric application package that has been generated by the build."
+        },
+        {
+            "name": "imageNamesPath",
+            "type": "filePath",
+            "label": "Image Names Path",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Path to a text file that contains the names of the Docker images to be pushed. Each image name is contained on its own line."
```

Comment:
Are we planing an update to our build definition to ensure this is saved as an artifact?